pkg/signature/kms/cliplugin/signer.go

```diff
@@ -144,10 +144,8 @@ func (c PluginClient) PublicKey(opts ...signature.PublicKeyOption) (crypto.Publi
 		},
 	}
 	ctx := context.Background()
-	if deadline := args.PublicKey.PublicKeyOptions.RPCOptions.CtxDeadline; deadline != nil {
-		var cancel context.CancelFunc
-		ctx, cancel = context.WithDeadline(ctx, *deadline)
-		defer cancel()
+	for _, opt := range opts {
+		opt.ApplyContext(&ctx)
```

Comment:
To confirm, is the expectation that the creator of the context will call `defer cancel()`?

Reply:
Yes, or they may call it `cancel()` early.